renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -3924,9 +3945,291 @@ void VulkanReplay::OverlayRendering::Init(WrappedVulkan *driver, VkDescriptorPoo
   }
 
+  samplesHandled = 0;
+  {
+    ConciseGraphicsPipeline DepthCopyPipeInfo = {
+        SRGBA8RP,
+        m_DepthCopyPipeLayout,
+        shaderCache->GetBuiltinModule(BuiltinShader::BlitVS),
+        shaderCache->GetBuiltinModule(BuiltinShader::DepthCopyFS),
+        {VK_DYNAMIC_STATE_VIEWPORT},
+        VK_SAMPLE_COUNT_1_BIT,
+        false,    // sampleRateShading
+        true,     // depthEnable
+        true,     // stencilEnable
+        VK_STENCIL_OP_ZERO,
+        true,     // colourOutput
+        false,    // blendEnable
+        VK_BLEND_FACTOR_DST_ALPHA,
+        VK_BLEND_FACTOR_ONE,
+        0x0,    // writeMask
+    };
+
+    VkAttachmentReference colRef = {0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL};
+    VkAttachmentReference dsRef = {1, VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL};
+
+    VkAttachmentDescription attDescs[] = {
+        {
+            0,
+            VK_FORMAT_R16G16B16A16_SFLOAT,
+            VK_SAMPLE_COUNT_1_BIT,
+            VK_ATTACHMENT_LOAD_OP_LOAD,
+            VK_ATTACHMENT_STORE_OP_STORE,
+            VK_ATTACHMENT_LOAD_OP_DONT_CARE,
+            VK_ATTACHMENT_STORE_OP_DONT_CARE,
+            colRef.layout,
+            colRef.layout,
+        },
+        {
+            0,
+            VK_FORMAT_D24_UNORM_S8_UINT,
+            VK_SAMPLE_COUNT_1_BIT,    // will patch this just below
+            VK_ATTACHMENT_LOAD_OP_DONT_CARE,
+            VK_ATTACHMENT_STORE_OP_STORE,
+            VK_ATTACHMENT_LOAD_OP_CLEAR,
+            VK_ATTACHMENT_STORE_OP_STORE,
+            dsRef.layout,
+            dsRef.layout,
+        },
+    };
+
+    VkSubpassDescription subp = {
+        0,      VK_PIPELINE_BIND_POINT_GRAPHICS,
+        0,      NULL,       // inputs
+        1,      &colRef,    // color
+        NULL,               // resolve
+        &dsRef,             // depth-stencil
+        0,      NULL,       // preserve
+    };
+
+    VkRenderPassCreateInfo rpinfo = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO,
+        NULL,
+        0,
+        2,
+        attDescs,
+        1,
+        &subp,
+        0,
+        NULL,    // dependencies
+    };
+
+    RDCCOMPILE_ASSERT(ARRAY_COUNT(m_DepthResolvePipeline) == ARRAY_COUNT(m_DepthResolvePipeline),
+                      "m_DepthCopyPipeline size must match m_DepthResolvePipeline");
+
+    if(DepthCopyPipeInfo.fragment != VK_NULL_HANDLE)
+    {
+      for(size_t f = 0; f < ARRAY_COUNT(m_DepthCopyPipeline); ++f)
+      {
+        VkFormat fmt = (f == 0) ? VK_FORMAT_D24_UNORM_S8_UINT : VK_FORMAT_D32_SFLOAT_S8_UINT;
+        attDescs[1].format = fmt;
+        for(size_t i = 0; i < ARRAY_COUNT(m_DepthCopyPipeline[f]); ++i)
+        {
+          m_DepthCopyPipeline[f][i] = VK_NULL_HANDLE;
+          VkSampleCountFlagBits samples = VkSampleCountFlagBits(1 << i);
+
+          if((supportedSampleCounts & (uint32_t)samples) == 0)
+            continue;
+
+          VkRenderPass DEPTHMSRP = VK_NULL_HANDLE;
```

Comment:
Uppercase for this variable name feels strange. I'd rather have `depthMSRP or `DepthMSRP` for a local like this.